vhdbuilder/packer/install-dependencies.sh

```diff
@@ -194,28 +194,49 @@ if [[ $OS == $UBUNTU_OS_NAME ]]; then
   done
 fi
 
+if [[ $OS == $UBUNTU_OS_NAME && $(isARM64) != 1 ]]; then  # no ARM64 SKU with GPU now
+  gpu_action="copy"
+  if grep -q "fullgpu" <<< "$FEATURE_FLAGS"; then
+    # force older driver on dedicated vhd to avoid build failures.
+    export NVIDIA_DRIVER_IMAGE_TAG="470.57.02-v0.0.2-rev17-sha-ff708e"
+    gpu_action="install"
+  else
+    export NVIDIA_DRIVER_IMAGE_TAG="510.47.03-v0.0.2-rev17-sha-ff708e"
+  fi
+
+  mkdir -p /opt/{actions,gpu}
+  if [[ "${CONTAINER_RUNTIME}" == "containerd" ]]; then
+    ctr image pull $NVIDIA_DRIVER_IMAGE:$NVIDIA_DRIVER_IMAGE_TAG
+    bash -c "$CTR_GPU_INSTALL_CMD $NVIDIA_DRIVER_IMAGE:$NVIDIA_DRIVER_IMAGE_TAG gpuinstall /entrypoint.sh $gpu_action" 
+    ret=$?
+    if [[ "$ret" != "0" ]]; then
+      echo "Failed to install GPU driver, exiting..."
+      exit $ret
+    fi
+    ctr c del gpuinstall
+    ctr images rm --sync $NVIDIA_DRIVER_IMAGE:$NVIDIA_DRIVER_IMAGE_TAG
+  else
+    bash -c "$DOCKER_GPU_INSTALL_CMD $NVIDIA_DRIVER_IMAGE:$NVIDIA_DRIVER_IMAGE_TAG $gpu_action"
+    ret=$?
+    if [[ "$ret" != "0" ]]; then
+      echo "Failed to install GPU driver, exiting..."
+      exit $ret
+    fi
+    docker rmi $NVIDIA_DRIVER_IMAGE:$NVIDIA_DRIVER_IMAGE_TAG 
+  fi
+fi
+
+ls -ltr /opt/gpu/* >> ${VHD_LOGS_FILEPATH}
+
 installBpftrace
 echo "  - bpftrace" >> ${VHD_LOGS_FILEPATH}
 
-if [[ $OS == $UBUNTU_OS_NAME && $(isARM64) != 1 ]]; then  # no ARM64 SKU with GPU now
-addNvidiaAptRepo
-installNvidiaDocker "${NVIDIA_DOCKER_VERSION}"
-downloadGPUDrivers
-retrycmd_if_failure 30 5 3600 wget "https://developer.download.nvidia.com/compute/cuda/redist/fabricmanager/linux-x86_64/fabricmanager-linux-x86_64-${GPU_DV}.tar.gz" || exit $ERR_GPU_DOWNLOAD_TIMEOUT
-tar -xvzf fabricmanager-linux-x86_64-${GPU_DV}.tar.gz -C /opt/azure
-mv /opt/azure/fabricmanager /opt/azure/fabricmanager-${GPU_DV}
-echo "  - nvidia-docker2 nvidia-container-runtime" >> ${VHD_LOGS_FILEPATH}
-downloadNvidiaContainerRuntime || exit $ERR_GPU_DRIVERS_INSTALL_TIMEOUT
-{
-  echo "  - nvidia-container-runtime=${NVIDIA_CONTAINER_RUNTIME_VERSION}";
-  echo "  - nvidia-gpu-driver-version=${GPU_DV}";
-  echo "  - nvidia-fabricmanager=${GPU_DV}";
-} >> ${VHD_LOGS_FILEPATH}
-if grep -q "fullgpu" <<< "$FEATURE_FLAGS"; then
-    echo "  - ensureGPUDrivers" >> ${VHD_LOGS_FILEPATH}
-    ensureGPUDrivers
-fi
-fi
+cat << EOF >> ${VHD_LOGS_FILEPATH}
+  - nvidia-docker2=${NVIDIA_DOCKER_VERSION}
+  - nvidia-container-runtime=${NVIDIA_CONTAINER_RUNTIME_VERSION}
+  - nvidia-gpu-driver-version=${GPU_DV}
```

Comment:
nit: should this have the major.minor.patch from NVIDIA_DRIVER_IMAGE_TAG, instead, because of the switch between the two versions? Or maybe I'm missing something.

Reply:
yeah, I wasn't sure if it was better here to list the container image version or the underlying components. I think keeping it like this is okay, but it means we have to sync these vars in cse_helpers.sh with the GPU image tags. a bit annoying. 

might have the installer export a file with this info we could pick up at build time instead